docs/runbooks/runbook-examples/routine/hardening-iis.md

```diff
@@ -0,0 +1,236 @@
+	 
+	New-Item 'HKLM:\SYSTEM\CurrentControlSet\Control\SecurityProviders\SCHANNEL\Hashes\SHA' -Force | Out-Null
+	New-ItemProperty -path 'HKLM:\SYSTEM\CurrentControlSet\Control\SecurityProviders\SCHANNEL\Hashes\SHA' -name Enabled -value '0xffffffff' -PropertyType 'DWord' -Force | Out-Null
+	 
+	# Set KeyExchangeAlgorithms configuration.
+	New-Item 'HKLM:\SYSTEM\CurrentControlSet\Control\SecurityProviders\SCHANNEL\KeyExchangeAlgorithms\Diffie-Hellman' -Force | Out-Null
+	New-ItemProperty -path 'HKLM:\SYSTEM\CurrentControlSet\Control\SecurityProviders\SCHANNEL\KeyExchangeAlgorithms\Diffie-Hellman' -name Enabled -value '0xffffffff' -PropertyType 'DWord' -Force | Out-Null
+	 
+	New-Item 'HKLM:\SYSTEM\CurrentControlSet\Control\SecurityProviders\SCHANNEL\KeyExchangeAlgorithms\PKCS' -Force | Out-Null
+	New-ItemProperty -path 'HKLM:\SYSTEM\CurrentControlSet\Control\SecurityProviders\SCHANNEL\KeyExchangeAlgorithms\PKCS' -name Enabled -value '0xffffffff' -PropertyType 'DWord' -Force | Out-Null
+	 
+	# Set cipher suites order as secure as possible (Enables Perfect Forward Secrecy).
+	$cipherSuitesOrder = @(
+	  'TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384_P521',
+	  'TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384_P384',
+	  'TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384_P256',
+	  'TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA_P521',
+	  'TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA_P384',
+	  'TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA_P256',
+	  'TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256_P521',
+	  'TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA_P521',
+	  'TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256_P384',
+	  'TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256_P256',
+	  'TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA_P384',
+	  'TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA_P256',
+	  'TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384_P521',
+	  'TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384_P384',
+	  'TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256_P521',
+	  'TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256_P384',
+	  'TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256_P256',
+	  'TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA384_P521',
+	  'TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA384_P384',
+	  'TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA_P521',
+	  'TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA_P384',
+	  'TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA_P256',
+	  'TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256_P521',
+	  'TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256_P384',
+	  'TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256_P256',
+	  'TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA_P521',
+	  'TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA_P384',
+	  'TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA_P256',
+	  'TLS_DHE_DSS_WITH_AES_256_CBC_SHA256',
+	  'TLS_DHE_DSS_WITH_AES_256_CBC_SHA',
+	  'TLS_DHE_DSS_WITH_AES_128_CBC_SHA256',
+	  'TLS_DHE_DSS_WITH_AES_128_CBC_SHA',
+	  'TLS_DHE_DSS_WITH_3DES_EDE_CBC_SHA',
+	  'TLS_RSA_WITH_AES_256_CBC_SHA256',
+	  'TLS_RSA_WITH_AES_256_CBC_SHA',
+	  'TLS_RSA_WITH_AES_128_CBC_SHA256',
+	  'TLS_RSA_WITH_AES_128_CBC_SHA',
+	  'TLS_RSA_WITH_3DES_EDE_CBC_SHA'
```

Comment:
This doesn't look right to me @twerthi 

`TLS_RSA_WITH_3DES_EDE_CBC_SHA`
`TLS_DHE_DSS_WITH_3DES_EDE_CBC_SHA`

AFAIK Triple Data Encryption Standard is not a perfect forward secrecy cipher suite and is no longer recommended by NIST

https://csrc.nist.gov/News/2017/Update-to-Current-Use-and-Deprecation-of-TDEA

The source of the script that you may have used appears to be a little bit different now https://www.hass.de/content/setup-microsoft-windows-or-iis-ssl-perfect-forward-secrecy-and-tls-12 ?

Reply:
Ah!  @wordlee , good call having @jburger review this!  @jburger Thanks for catching that, I've removed the triple des stuff and submitted another PR.  I did find this buried in the comments of the script, `# - 3DES: It is recommended to disable these in near future` So it looks like it is just out of date.  

In regards to your question about whether regularly updating it, I didn't really have any plans.  This was to be provided as an example of what is possible.  I'm not against keeping it up to date, but not sure where the responsibility would lie.  I'm certainly up for discussion.

Reply:
> In regards to your question about whether regularly updating it, I didn't really have any plans. This was to be provided as an example of what is possible. I'm not against keeping it up to date, but not sure where the responsibility would lie. I'm certainly up for discussion.

We don't currently have any formal processes around updating content. My preference is to include the security disclaimer from #857, and make sure users understand, the guidance is for demonstration purposes only, and they should check ... still think of the words :)